### PR TITLE
WIP: Metrics in MonitorCrossConnects

### PR DESCRIPTION
**Pull request by platoff**
Implementation of Metrics in MonitorCrossConnects/MonitorConnections #715 

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/monitor/monitor_server.go [hunk 1]
```diff
@@ -12,6 +14,11 @@ const (
 	INITIAL_STATE_TRANSFER = "INITIAL_STATE_TRANSFER"
 )
 
+const (
+	src_prefix = "SRC-"
+	dst_prefix = "DST-"
```
Comment (nickolaev) [inline controlplane/pkg/monitor/monitor_server.go]: Where these come from? We should maintain common constant definitions across all packages we generate. Please consider refactoring the other modules if needed.

File: controlplane/pkg/monitor/monitor_server.go [hunk 2]
```diff
@@ -21,6 +28,11 @@ type Event struct {
 	Entities  map[string]Entity
 }
 
+type Stat struct {
```
Comment (nickolaev) [inline controlplane/pkg/monitor/monitor_server.go]: Can we have a more elaborate name of the structure here? `Stat` is pretty short. Also do we need this type exported?

File: controlplane/pkg/monitor/monitor_server.go [hunk 3]
```diff
@@ -33,6 +45,8 @@ type MonitorServer interface {
 	Update(entity Entity)
 	Delete(entity Entity)
 
+	UpdateStat(name string, statistics *interfaces.InterfacesState_Interface_Statistics)
```
Comment (nickolaev) [inline controlplane/pkg/monitor/monitor_server.go]: I believe this should be plural: `UpdateStats` or even better `UpdateStatistics`

Comment (platoff) [reply controlplane/pkg/monitor/monitor_server.go]: @nickolaev what do you think about bad abstractions here. I'd prefer to remove monitor server and use concrete implementation instead. Any thoughts on this?

File: controlplane/pkg/monitor/monitor_server.go (hunk 1 above)
Comment (edwarnicke) [reply controlplane/pkg/monitor/monitor_server.go]: If these are related to the prefixes we happened to have selected in vppagent-dataplane... I'd prefer *not* to presume upon them, and they may at some point change...

File: controlplane/pkg/monitor/monitor_server.go (hunk 3 above)
Comment (edwarnicke) [reply controlplane/pkg/monitor/monitor_server.go]: Or possibly UpdateMetrics ? (brainstorming :) )

**Comment by nickolaev** [comment]
#980 supersedes this one. Closing.

**Pull request closed without merging**